packages/contracts-bedrock/src/L2/SoulGasToken.sol

```diff
@@ -0,0 +1,197 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+import { ERC20Upgradeable } from "@openzeppelin/contracts-upgradeable/token/ERC20/ERC20Upgradeable.sol";
+import { OwnableUpgradeable } from "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+import { Constants } from "src/libraries/Constants.sol";
+
+/// @title SoulGasToken
+/// @notice The SoulGasToken is a soul-bounded ERC20 contract which cab be used to pay gas on L2.
+///         It has 2 modes:
+///             1. when Constants.IS_SOUL_QKC(or in other words: SoulQKC mode), the token can be minted by anyone
+///                depositing native token into the contract.
+///             2. when !Constants.IS_SOUL_QKC(or in other words: SoulETH mode), the token can only be minted by
+///                whitelist minters specified by contract owner.
+contract SoulGasToken is ERC20Upgradeable, OwnableUpgradeable {
+    /// @custom:storage-location erc7201:openzeppelin.storage.SoulGasToken
+    struct SoulGasTokenStorage {
+        // _minters are be whitelist EOAs, only used when !Constants.IS_SOUL_QKC
+        mapping(address => bool) _minters;
+        // _burners are whitelist EOAs, only used when !Constants.IS_SOUL_QKC
+        mapping(address => bool) _burners;
+    }
+
+    // keccak256(abi.encode(uint256(keccak256("openzeppelin.storage.SoulGasToken")) - 1)) & ~bytes32(uint256(0xff))
+    bytes32 private constant _SOULGASTOKEN_STORAGE_LOCATION =
+        0x135c38e215d95c59dcdd8fe622dccc30d04cacb8c88c332e4e7441bac172dd00;
+
+    function _getSoulGasTokenStorage() private pure returns (SoulGasTokenStorage storage $) {
+        assembly {
+            $.slot := _SOULGASTOKEN_STORAGE_LOCATION
+        }
+    }
+
+    constructor() {
+        _disableInitializers();
+    }
+
+    /// @custom:legacy
+    /// @notice initialize is used to initialize SoulGasToken contract.
+    function initialize(
+        string calldata name_,
+        string calldata symbol_,
+        address owner_,
+        address[] calldata minters_,
+        address[] calldata burners_
+    )
+        external
+        initializer
+    {
+        if (Constants.IS_SOUL_QKC) {
```

Comment:
Since we already rely on `owner_` to set minters_ and burners_, I think we can remove minters_ and burners_ parameters and set them later by owner

Reply:
Good point!